### PR TITLE
gh-104454: Fix refleak in AttributeError_reduce

### PR DESCRIPTION
**Pull request by csm10495**
gh-104454: Fix ref leak introduced by gh-103333

### CODE REVIEW

**Comment by csm10495** [comment]
(i'm not sure how to apply skip-news.. but don't think this needs a news entry)

File: Objects/exceptions.c [hunk 1]
```diff
@@ -2324,7 +2324,9 @@ AttributeError_reduce(PyAttributeErrorObject *self, PyObject *Py_UNUSED(ignored)
         return NULL;
     }
 
-    return PyTuple_Pack(3, Py_TYPE(self), self->args, state);
+    PyObject* return_value = PyTuple_Pack(3, Py_TYPE(self), self->args, state);
```
Comment (Eclips4) [inline Objects/exceptions.c]:
```suggestion
    PyObject *return_value = PyTuple_Pack(3, Py_TYPE(self), self->args, state);
```
For consistency with the rest of function =)

**Review by Eclips4 — APPROVED** [review]
I rebuilt interpreter with this changes, and refleaks are gone.
Thanks for fix!

**Review by hauntsaninja — APPROVED** [review]
Looks good, thanks for fixing!

**Review by iritkatriel — APPROVED** [review]

**Comment by gpshead** [comment]
Thanks!

**Comment by AlexWaygood** [comment]
> Failed tests:
> 
> * test__xxsubinterpreters

This is tracked by https://github.com/python/cpython/issues/104341, and is unrelated to this PR.

**Pull request merged**